code/controllers/subsystem/non_firing/SSlate_mapping.dm

```diff
@@ -27,4 +27,30 @@ SUBSYSTEM_DEF(late_mapping)
 		var/duration = stop_watch(watch)
 		log_startup_progress("Generated [mgcount] mazes in [duration]s")
 
+	maintenance_mice()
+
 	GLOB.spawn_pool_manager.process_pools()
+
+/**
+ * Randomly spawns mice in maintenance instead of being purely fixed spawn points
+ */
+/datum/controller/subsystem/late_mapping/proc/maintenance_mice()
+	var/watch = start_watch()
+	log_startup_progress("Populating maintenance with mice...")
+
+	// Looking up for maintenance floors specifically as possible spawn points
+	var/list/maintenance_turfs = list()
+	for(var/area/station/maintenance/A in SSmapping.existing_station_areas)
+		for(var/turf/simulated/floor/F in A)
+			if(locate(/obj/structure/window) in F)
+				continue
+			maintenance_turfs.Add(F)
+	
+	// The ratio is based on turfs per mice. Using Boxstation as an example, it would average between 20 to 30 mice.
+	var/ratio = rand(125, 200)
+	var/mice_number = floor(length(maintenance_turfs) / ratio)
+
+	for(var/i in 1 to mice_number)
+		new /mob/living/simple_animal/mouse(pick_n_take(maintenance_turfs))
+
+	log_startup_progress("Spawned [mice_number] mice over in [stop_watch(watch)]s")
```

Comment:
could you move this to a debug statement so that players aren't told how many mice spawn?